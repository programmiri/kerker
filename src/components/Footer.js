import React from "react";

function Footer(props) {
  return (
    <footer className="Footer">
       <small className="font-s">
        made with <i className="fas fa-heartbeat" /> and <i className="fas fa-coffee" />
      </small>
    </footer>
  );
}

export default Footer;
