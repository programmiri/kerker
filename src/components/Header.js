import React from "react";
import logo from "./../images/logo-color.png";

function Header(props) {
  return (
    <div className="Header d-flex justify-content-end align-items-center">
      <h1 className="Header-text text-secondary pr-3">Welcome to Kerker</h1>
      <img src={logo} className="Header-logo" alt="logo" />
    </div>
  );
}

export default Header;
