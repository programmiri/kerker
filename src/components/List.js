import React from "react";
import ListItem from "./ListItem";
import PropTypes from "prop-types";

function List(props) {
  const listItems = props.notes.map((item, index) => (
    <ListItem
      key={item.id}
      id={item.id}
      title={item.title}
      createdAt={item.createdAt}
      note={item}
      setCurrentNote={props.setCurrentNote}
      setCurrentNoteEncodingState={props.setCurrentNoteEncodingState}
      index={index}
      isActive={props.currentNote === item}
    />
  ));
  return <ul className="List list-group">{listItems}</ul>;
}

export default List;

List.propTypes = {
  notes: PropTypes.array.isRequired
};
