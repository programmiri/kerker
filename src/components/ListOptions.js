import React from "react";

function ListOptions(props) {
  return (
    <div className="ListOptions text-right pr-1">
      <button className="ListOptions-sorting mr-2" href="" alt="sort by date asc">
        <i className="fas fa-2x fa-angle-up" />
      </button>
      <button className="ListOptions-sorting" href="" alt="sort by date desc">
        <i className="fas fa-2x fa-angle-down" />
      </button>
    </div>
  );
}

export default ListOptions;
