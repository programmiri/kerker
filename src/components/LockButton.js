import React from "react";

function LockButton(props) {


  const toggleEncodingState = event => {
    event.preventDefault();
    props.setCurrentNoteEncodingState(!props.currentNoteEncrypted);
  };

  const buttonIcon = props.currentNoteEncrypted ? "fa-unlock" : "fa-lock";

  return (
    <div className="LockButton text-right pb-4" onClick={toggleEncodingState}>
      <i className={`fas fa-3x ${buttonIcon}`} />
    </div>
  );
}

export default LockButton;
