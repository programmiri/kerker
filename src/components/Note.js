import React from "react";

function Note(props) {
  return (
    <div className="">
      <p>
        {props.text}
      </p>
    </div>
  );
}

export default Note;
