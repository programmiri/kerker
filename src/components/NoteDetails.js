import React from "react";
import Alert from "./Alert";
import { dateToString } from "../lib/format.js";

function NoteDetails(props) {
  return (
    <React.Fragment>
      <Alert context="warning" text="Dont't forget to decrypt your note!"/>
      <small className="font-s  font-monospace">
        <i className="far fa-calendar" /> created: {dateToString(props.createdAt)}
      </small>
    </React.Fragment>
  );
}

export default NoteDetails;
