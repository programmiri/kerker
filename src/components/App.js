import React, { Component } from "react";
import Footer from "./Footer";
import Header from "./Header";
import List from "./List";
import ListOptions from "./ListOptions";
import Note from "./Note";
import NoteDetails from "./NoteDetails";
import LockButton from "./LockButton";
import NotePlaceholder from "./NotePlaceholder";

class App extends Component {
  renderLogin() {
    return <div className="App-login-notice">You're not logged in</div>;
  }

  renderNote() {
    const noteDetails = this.props.currentNoteEncrypted ?  <NoteDetails createdAt={this.props.currentNote.createdAt} /> : null;
    const note = this.props.currentNoteEncrypted ? <Note text={this.props.currentNote.bodyPersistedVersion}/> : <NotePlaceholder />;
    return (
      <div className="App-note-section ml-4">
        <LockButton currentNoteEncrypted={this.props.currentNoteEncrypted} setCurrentNoteEncodingState={this.props.setCurrentNoteEncodingState}/>
        {noteDetails}
        {note}
      </div>
    );
  }

  renderNotePlaceholder() {
    return (
      <div className="App-note-section ml-4 d-flex align-items-center justify-content-center">
        Currently no note selected.
      </div>
    );
  }

  renderNoteSection() {
    const isNoteSelected = this.props.currentNote;
    const currentNote = isNoteSelected
      ? this.renderNote()
      : this.renderNotePlaceholder();

    return (
      <React.Fragment>
        <div className="App-note-options">
          <ListOptions />
          <List
            notes={this.props.notes}
            currentNote={this.props.currentNote}
            setCurrentNote={this.props.setCurrentNote}
            setCurrentNoteEncodingState={this.props.setCurrentNoteEncodingState}
          />
        </div>
        {currentNote}
      </React.Fragment>
    );
  }

  renderMainContent() {
    return this.props.userIsLoggedIn
      ? this.renderNoteSection()
      : this.renderLogin();
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <Header positionInGrid=""/>
        </header>
        {this.renderMainContent()}
        <Footer />
      </div>
    );
  }
}

export default App;
