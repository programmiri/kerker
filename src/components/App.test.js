import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import App from "./App";
import Footer from "./Footer";
import Header from "./Header";
import List from "./List";
import ListOptions from "./ListOptions";
import Note from "./Note";
import NoteDetails from "./NoteDetails";
import LockButton from "./LockButton";

const notesFixtures = [
  {
    id: "109156be-c4fb-41ea-b1b4-efe1671c5836",
    title: "Hello",
    body: "This is my body",
    bodyPersistedVersion: "This is my body",
    bodyEncrypted: "SECRET",
    createdAt: new Date(1517743704 * 1000),
    updatedAt: new Date(1517743704 * 1000)
  },
  {
    id: "209156be-c4fb-41ea-b1b4-efe1671c5836",
    title: "Another note",
    body: "Another body",
    bodyPersistedVersion: "Another body",
    bodyEncrypted: "SECRET2",
    createdAt: new Date(1517743704 * 1000),
    updatedAt: new Date(1517743704 * 1000)
  }
];

const currentNoteFixture = notesFixtures[0];

describe("<App /> when not logged in", () => {
  it("renders without crashing", () => {
    shallow(<App notes={notesFixtures} userIsLoggedIn={false} />);
  });

  const wrapper = shallow(<App notes={notesFixtures} userIsLoggedIn={false} />);

  it("does not renders <Notes />", () => {
    expect(wrapper.find(Note).exists()).toBe(false);
  });

  it("snapshot test", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

describe("<App /> when logged in", () => {
  it("renders without crashing", () => {
    shallow(<App notes={notesFixtures} userIsLoggedIn={true} />);
  });

  it("renders always <ListOptions /> and <List /> components", () => {
    const wrapper = shallow(
      <App notes={notesFixtures} userIsLoggedIn={true} />
    );
    expect(wrapper.find(ListOptions).exists()).toBe(true);
    expect(wrapper.find(List).exists()).toBe(true);
  });

  describe("renders <Note /> and <NoteDetails /> only when a Note is selcted", () => {
    describe("renders shows an placeholder as long as there is no note selected", () => {
      const wrapper = shallow(
        <App notes={notesFixtures} userIsLoggedIn={true} currentNote={null} />
      );

      it("renders shows an placeholder as long as there is no note selected", () => {
        expect(wrapper.find(NoteDetails).exists()).toBe(false);
        expect(wrapper.find(Note).exists()).toBe(false);
        expect(wrapper.contains("No note yet.")).toBe(true);
      });

      it("snapshot test", () => {
        expect(toJson(wrapper)).toMatchSnapshot();
      });
    });

    describe("renders shows an placeholder as long as there is no note selected", () => {
      const wrapper = shallow(
        <App
          notes={notesFixtures}
          userIsLoggedIn={true}
          currentNote={currentNoteFixture}
        />
      );
      it("renders <Note /> and <NoteDetails /> when a Note is selcted", () => {
        expect(wrapper.find(NoteDetails).exists()).toBe(true);
        expect(wrapper.find(Note).exists()).toBe(true);
      });

      it("snapshot test", () => {
        expect(toJson(wrapper)).toMatchSnapshot();
      });
    });
  });
});
