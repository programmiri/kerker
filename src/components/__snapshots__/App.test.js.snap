// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> when logged in renders <Note /> and <NoteDetails /> only when a Note is selcted renders shows an placeholder as long as there is no note selected snapshot test 1`] = `
<div
  className="App"
>
  <Header
    positionInGrid="header"
  />
  <React.Fragment>
    <div
      className="App-note-options pr-5"
    >
      <ListOptions />
      <List
        notes={
          Array [
            Object {
              "body": "This is my body",
              "bodyEncrypted": "SECRET",
              "bodyPersistedVersion": "This is my body",
              "createdAt": 2018-02-04T11:28:24.000Z,
              "id": "109156be-c4fb-41ea-b1b4-efe1671c5836",
              "title": "Hello",
              "updatedAt": 2018-02-04T11:28:24.000Z,
            },
            Object {
              "body": "Another body",
              "bodyEncrypted": "SECRET2",
              "bodyPersistedVersion": "Another body",
              "createdAt": 2018-02-04T11:28:24.000Z,
              "id": "209156be-c4fb-41ea-b1b4-efe1671c5836",
              "title": "Another note",
              "updatedAt": 2018-02-04T11:28:24.000Z,
            },
          ]
        }
      />
    </div>
    <div
      className="App-note-section font-hint bg-light mb-3 d-flex justify-content-center align-items-center"
    >
      Currently no note selected.
    </div>
  </React.Fragment>
  <Footer />
</div>
`;

exports[`<App /> when logged in renders <Note /> and <NoteDetails /> only when a Note is selcted renders shows an placeholder as long as there is no note selected snapshot test 2`] = `
<div
  className="App"
>
  <Header
    positionInGrid="header"
  />
  <React.Fragment>
    <div
      className="App-note-options pr-5"
    >
      <ListOptions />
      <List
        notes={
          Array [
            Object {
              "body": "This is my body",
              "bodyEncrypted": "SECRET",
              "bodyPersistedVersion": "This is my body",
              "createdAt": 2018-02-04T11:28:24.000Z,
              "id": "109156be-c4fb-41ea-b1b4-efe1671c5836",
              "title": "Hello",
              "updatedAt": 2018-02-04T11:28:24.000Z,
            },
            Object {
              "body": "Another body",
              "bodyEncrypted": "SECRET2",
              "bodyPersistedVersion": "Another body",
              "createdAt": 2018-02-04T11:28:24.000Z,
              "id": "209156be-c4fb-41ea-b1b4-efe1671c5836",
              "title": "Another note",
              "updatedAt": 2018-02-04T11:28:24.000Z,
            },
          ]
        }
      />
    </div>
    <div
      className="App-note-section border p-5 mb-3"
    >
      <LockButton
        currentNote={
          Object {
            "body": "This is my body",
            "bodyEncrypted": "SECRET",
            "bodyPersistedVersion": "This is my body",
            "createdAt": 2018-02-04T11:28:24.000Z,
            "id": "109156be-c4fb-41ea-b1b4-efe1671c5836",
            "title": "Hello",
            "updatedAt": 2018-02-04T11:28:24.000Z,
          }
        }
      />
      <NoteDetails />
      <Note />
    </div>
  </React.Fragment>
  <Footer />
</div>
`;

exports[`<App /> when not logged in snapshot test 1`] = `
<div
  className="App"
>
  <Header
    positionInGrid="header"
  />
  <div
    className="App-login-notice"
  >
    You're not logged in
  </div>
  <Footer />
</div>
`;
